dev/nix/shell.nix

```diff
@@ -15,140 +15,63 @@
-        gawk
-        cargo-edit
-        perl
-      ]
-      ++ (
-        if isDarwin
-        then
-           [ pkgs.darwin.apple_sdk.frameworks.SystemConfiguration ]
-        else [ pkgs.clang ]
-      );
-    env = [
-      {
-        name = "RUST_SRC_PATH";
-        value = "${rustToolchain}/lib/rustlib/src/rust/library";
-      }
-      { name = "LIBCLANG_PATH"; value = "${pkgs.libclang.lib}/lib"; }
-      { name = "LD_LIBRARY_PATH"; value = "${rustToolchain}/lib"; }
-      { name = "ROCKSDB_LIB_DIR"; value = "${pkgs.rocksdb}/lib/"; }
-      { name = "OPENSSL_NO_VENDOR"; value = 1; }
-      { name = "OPENSSL_DIR"; value = "${pkgs.openssl.dev}"; }
-      { name = "OPENSSL_INCLUDE_DIR"; value = "${pkgs.openssl.dev}/include"; }
-      { name = "OPENSSL_LIB_DIR"; value = "${pkgs.openssl.out}/lib"; }
+  in {
+    devShells = {
+      default = pkgs.mkShell {
+        packages = with pkgs; [
+          coreutils
+          pkg-config
+          protobuf
 
-    ];
-    # Main Categories which can include pkgs, or devShell-like sets
-    # for commands and helpers
-    commands = self.lib.categorize [
-      {
-        help = "Earthly, an easy to use CI tool";
-        name = "earthly";
-        package = pkgs.earthly;
-        category = "CI/CD";
-        pkgs = [
-          pkgs.earthly
-          pkgs.awscli2
-          pkgs.kubectl
-          pkgs.kubernetes-helm
-        ];
-      }
-      {
-        category = "Rust";
-        pkgs = [
-          {
-            name = "check";
-            help = "Check rustc and clippy warnings";
-            command = ''
-              set -x
-              cargo check --all-targets
-              cargo clippy --all-targets
-            '';
-          }
-          {
-            help = "Automatically fix rustc and clippy warnings";
-            name = "fix";
-            command = ''
-              set -x
-              cargo fix --all-targets --allow-dirty --allow-staged
-              cargo clippy --all-targets --fix --allow-dirty --allow-staged
-            '';
-          }
-        ];
-      }
-      {
-        category = "Partner Chains";
-        pkgs = [
-          {
-            name = "cardano-cli";
-            help = "CLI v10.1.2 that is used in partner-chains dependency stack";
-            # This command has some eval because of IFD
-            command = "${self'.packages.cardano-cli}/bin/cardano-cli latest $@";
-          }
+          rustToolchain
+          nodejs
+          nodePackages.npm
+          pkgs.libiconv
+          pkgs.openssl
+          gnumake
+          gawk
+          cargo-edit
+          perl
+
+          earthly
+          awscli2
+          kubectl
+          kubernetes-helm
+
+          self'.packages.cardano-cli
+          self'.packages.pc-contracts-cli
+        ]
+        ++ (
+          if isDarwin
+          then
+            [ pkgs.darwin.apple_sdk.frameworks.SystemConfiguration ]
+          else [ pkgs.clang ]
+        );
+        shellHook = ''
```

Comment:
stylistic choice, but instead of using self hook for setting vars, the args passed to `mkShell` (its really `mkDerivation` underneath) are set in the environment automatically.

Up to you, but I typically prefer that unless there is some reason I actually need the hook, like some shell logic to generate a value, etc.

Reply:
oh yeah, I prefer that way actually. getting back from PTO i just blanked my mind completely. thanks. shellHooks can be for other things